### PR TITLE
feat(task): support --filter

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Closes https://github.com/denoland/deno/issues/24991

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -2951,13 +2953,27 @@ Evaluate a task from string
...
+          .long("recursive")
+          .short('r')
+          .help("Run the task in all projects in the workspace")
+          .action(ArgAction::SetTrue),
+      )
+      .arg(
+        Arg::new("filter")
+        .long("filter")
+        .short('f')
+        .help("Filter members of the workspace by name - should be used with --recursive")
```
Comment (nathanwhit) [inline cli/args/flags.rs:2967]:
```suggestion
        .help("Filter members of the workspace by name - should be used with --recursive")
        .requires("recursive")
```

Comment (nathanwhit) [reply cli/args/flags.rs:2967]: actually, maybe we should make `--filter` imply recursive

**Review by nathanwhit — COMMENTED** [review]
i think we should actually make --filter imply --recursive (i believe that's what pnpm does)

File: tests/specs/task/filter/deno_workspace_order.out [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Task dev echo '@foo/baz'
-@foo/baz
 Task dev echo '@foo/bar'
 @foo/bar
+Task dev echo '@foo/baz'
```
Comment (marvinhagemeister) [inline tests/specs/task/filter/deno_workspace_order.out:3]: Note to self: I need to take a look at these tomorrow. I think the order of tasks is wrong here but maybe I made it too confusing with the fixtures.

Comment (bartlomieju) [reply tests/specs/task/filter/deno_workspace_order.out:3]: They didn't seem to depend on each other, and I believe were printed in reversed order

File: cli/args/flags.rs (hunk 1 above)
Comment (dsherret) [reply cli/args/flags.rs:2967]: This broke the tests so I reverted it. We should resolve this before landing.

File: cli/tools/task.rs [hunk 3]
```diff
@@ -55,18 +64,149 @@ pub async fn execute_script(
...
+    }
+
+    // Logging every task definition would be too spammy. Pnpm only
+    // logs a simple message too.
+    if packages_task_info
+      .iter()
+      .all(|config| config.matched_tasks.is_empty())
+    {
+      writeln!(
+        &mut std::io::stdout(),
```
Comment (dsherret) [inline cli/tools/task.rs:176]: @marvinhagemeister for the future, we can use `log::warn` instead (which allows people to disable via `--quiet` or `--log-level` and also logs to stderr.

File: tests/specs/task/filter/deno_workspace_order.out (hunk 2 above)
Comment (marvinhagemeister) [reply tests/specs/task/filter/deno_workspace_order.out:3]: The tasks itself yeah, but the packages do. When package A depends on B, the matching tasks from B must be executed first.

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/args/flags.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/args/flags.rs:2967]: Changed it to make `--filter` imply `--recursive`.

**Pull request merged**